packages/peregrine/lib/talons/SignIn/useSignIn.js

```diff
@@ -1,25 +1,28 @@
-import { useCallback, useRef, useState } from 'react';
+import { useCallback, useEffect, useRef, useState } from 'react';
 import { useUserContext } from '../../context/user';
-import { useMutation } from '@apollo/react-hooks';
+import { useMutation, useLazyQuery } from '@apollo/react-hooks';
 import { useCartContext } from '../../context/cart';
 
 export const useSignIn = props => {
     const {
+        getCustomerQuery,
         setDefaultUsername,
         showCreateAccount,
         showForgotPassword,
-        query
+        signInMutation
     } = props;
 
     const [isSigningIn, setIsSigningIn] = useState(false);
 
     const [, { getCartDetails, removeCart }] = useCartContext();
-    const [
-        { isGettingDetails, getDetailsError },
-        { getUserDetails, setToken }
-    ] = useUserContext();
+    const [, { actions: userActions, setToken }] = useUserContext();
 
-    const [signIn, { error: signInError }] = useMutation(query);
+    const [signIn, { error: signInError }] = useMutation(signInMutation);
+    const [
+        getUserDetails,
+        { called, data: getCustomerData, error: getDetailsError, loading }
+    ] = useLazyQuery(getCustomerQuery);
```

Comment:
I wonder if we even need to query for this. It is possible the mutation returns what we need to render.

Reply:
This mutation unfortunately only returns the token, customer data wasn't available.

Reply:
Ok, what about doing something similar to what I've done in the `createCart` PR?
Basically we have some global/app effect that runs whenever some id necessary to get data is modified.

https://github.com/magento/pwa-studio/pull/1988/files#diff-4080b54df336f2a5838a43c80bc67391R46-R62